control/control.go

```diff
@@ -906,23 +982,79 @@ func (p *pluginControl) sendPluginSubscriptionEvent(taskID string, pl core.Plugi
 
 func (p *pluginControl) UnsubscribeDeps(taskID string, mts []core.Metric, plugins []core.Plugin) []serror.SnapError {
```

Comment:
Would you mind to add some comments how UnsubscribeDeps works? Based on the name of this function and its arguments I assumed that input mts and plugins will be unsubscribed, what is incorrectly. By the way, thank you @marcin-krolik that you explained me that this function will unsubscribe everything what does not meet with incoming mts and plugins.